packages/api/lib/SemaphoreUtils.js

```diff
@@ -9,23 +9,23 @@ const Semaphore = require('./Semaphore');
  * Throws `ResourcesLockedError` if maximum number of executions are already
  * running.
  *
- * @param {string} queueName - Queue name which is used as the semaphore key
+ * @param {string} queueUrl - Queue URL which is used as the semaphore key
  * @param {number} maximum - Maximum number of executions allowed for this semaphore
  * @returns {Promise}
  * @throws {Error}
  */
-async function incrementQueueSemaphore(queueName, maximum) {
+async function incrementQueueSemaphore(queueUrl, maximum) {
```

Comment:
Is this data migrated on deploy, so we don't have to worry about migration/backwards compat?

Reply:
There's no current data migration in place. I don't know how easy one would even be to create - because we would have to translate the current `queueName` key (which isn't actually an SQS queue name but an arbitrary identifier) to a queue URL. 

Once users upgrade to the version with these changes - they'll have to update how their workflows/rules configure execution throttling to use queue URLs, which is called out in the changelog migration steps. If you deploy without taking the migration steps, I would expect workflow failures. If you do deploy after taking the migration steps, then the execution limits based on new config should immediately take effect. Either way, I would expect that the execution throttling using semaphore keys based on queue name should keep working right up until you re-deploy, so do we need to migrate?

Reply:
I guess I'm asking several questions:

1. Given that the queueUrl is in the message with correctly configured limits, does this work? (assuming yes)
2. What do the users need to do to make sure their message is correct to work as per 1?
3. Does any of this respecting existing entries in the semaphores table? (assuming no)
4. If no to 3, is there any way we prevent a throttled workflow from having up to 2n-1 executions, where n is the limit, if there were n executions at the time of deploying the upgrade?
5. If no to 4, should we advise users to stop throttled workflow executions before deploying the upgrade?

Reply:
1. If this means does the execution throttling work if `cumulus_meta.queueUrl` and `cumulus_meta.queueExecutionLimits` are set in the message, then the answer is yes which is confirmed by manual and automated testing
2. In order to configure the message, you just need to follow the instructions in the [updated docs](https://github.com/nasa/cumulus/pull/1782/files#diff-f5ad827a5ab8f6085a8529e9a99ff437) for either using rules or updating the queue for your queue-granules/queue-pdrs steps
3. No, it would not respect existing entries in the table
4. Without a migration, I don't see a way to prevent it. If the only harm is that there are temporarily up to 2n-1 executions but then no more than `n` once executions based on the old semaphore key have finished, is that a big deal?
5. If we don't migrate the existing keys and we determine that the adverse effect of 4 is high enough, then I guess so. Most likely, that means they'd have to disable rules running their discover workflows or other rules using a throttled queue. Is this really feasible?

Reply:
I'd like to document *something* to call out the effects of 4. It's not likely to be a huge issue, but in theory since step functions can run up to a year, you could have more workflows than you want running out there for a very long time.

Reply:
We _could_ write a script, but it would have to run pre-upgrade in order to convert your semaphore keys to queue URLs. And I'm not sure how we would provide it to users. Thoughts?

Reply:
I think informing them will do. Put a clear note in the breaking changes section of the changelog, and point out that they may have double the number of expected executions. If it becomes an issue, we can address it then rather than spend time pre-emptively creating a script for an audience of possibly no one.

Reply:
c005bd9224b115c1f92be69dad73bdf0000af9d4